### PR TITLE
add more html pages

### PR DESCRIPTION
**Pull request by maksimk99**
1. add list car features to add new advertisement page
2. add page of car sale advertisement
3. add login page

### CODE REVIEW

File: src/main/java/com/carhouse/controller/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+/**
+ * Login controller.
+ */
+@Controller
+public class LoginController {
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/controller/LoginController.java:14]: Do you have the requirement to create login page?

File: src/main/java/com/carhouse/provider/impl/CarMakeProviderImpl.java [hunk 2]
```diff
@@ -34,8 +34,12 @@
      * @return the car make
      */
     public CarMake getCarMake(final String carMakeId) {
-        if (carMakeId != null && Integer.parseInt(carMakeId) >= 0) {
+        if (carMakeId != null && !carMakeId.isEmpty()) {
+            try {
             return new CarMake(Integer.parseInt(carMakeId), "BMW");
+            } catch (NumberFormatException ex) {
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/provider/impl/CarMakeProviderImpl.java]: looks strange.
Are you pretty confident that it is the best way how to validate `carMakeId`?

File: src/main/java/com/carhouse/controller/LoginController.java (hunk 1 above)
Comment (maksimk99) [reply src/main/java/com/carhouse/controller/LoginController.java:14]: I had a time so I decided to make pages what I will need in a feature.

Comment (aleksieiev) [reply src/main/java/com/carhouse/controller/LoginController.java:14]: Discussed

File: src/main/java/com/carhouse/provider/CarSaleProvider.java [hunk 3]
```diff
@@ -17,9 +17,21 @@
...
     /**
      * Add new car sale advertisement.
-     * @param carSale object from form
+     * Take car features id and set them to the car sale object
+     *
+     * @param carSale     object from form
+     * @param carFeatures list of selected car feature's id
      */
-    void addCarSale(CarSale carSale);
+    void addCarSale(CarSale carSale, int[] carFeatures);
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/provider/CarSaleProvider.java:36]: Is it array? Rly?

**Review by aleksieiev — APPROVED** [review]

**Pull request merged**